web/scenes/Portal/Profile/common/UserInfo/index.tsx

```diff
@@ -19,41 +21,48 @@ export const UserInfo = (props: UserInfoProps) => {
   const userHasura = user?.hasura;
   const userId = userHasura?.id;
 
-  const userQueryRes = useFetchUserQuery({
+  const { data } = useFetchUserQuery({
     variables: !userId ? undefined : { user_id: userId },
     context: {
       headers: { team_id: "_" },
     },
     skip: !userId,
   });
 
+  const name = useMemo(
+    () =>
+      props.name ||
+      data?.user?.name ||
+      data?.user?.email ||
+      getNullifierName(data?.user?.world_id_nullifier) ||
+      "Anonymous User",
+    [
+      data?.user?.email,
+      data?.user?.name,
+      data?.user?.world_id_nullifier,
+      props.name,
+    ],
+  );
+
   return (
     <div
       className={twMerge(clsx("flex items-center gap-x-5", props.className))}
     >
       <div className="leading-[0]">
-        {!userQueryRes.data?.user ? (
+        {!data?.user ? (
           <Skeleton width={72} height={72} circle={true} inline={true} />
         ) : (
-          <Icon name={props.name ?? userQueryRes.data?.user?.name} />
+          <Icon name={name} />
         )}
       </div>
 
       <div className="grid gap-y-2">
         <Typography variant={TYPOGRAPHY.H6}>
-          {!userQueryRes.data?.user ? (
-            <Skeleton width={200} />
-          ) : (
-            props.name ?? userQueryRes.data.user.name
-          )}
+          {!data?.user ? <Skeleton width={200} /> : name}
         </Typography>
 
         <Typography variant={TYPOGRAPHY.R4} className="text-grey-500">
-          {!userQueryRes.data?.user ? (
-            <Skeleton width={200} />
-          ) : (
-            userQueryRes.data.user.email || null
-          )}
+          {!data?.user ? <Skeleton width={200} /> : data.user.email || null}
```

Comment:
```suggestion
          {!data?.user ? <Skeleton width={200} /> : data.user?.email || null}
```
Will break if email is null?